calicoctl/commands/node/run.go

```diff
@@ -181,15 +194,23 @@ Description:
 
 	// Create a mapping of environment variables to values.
 	envs := map[string]string{
-		"HOSTNAME": name,
```

Comment:
The Felix runit job expects HOSTNAME to be set I think.  It sets the FELIX_HOSTNAME value based on that.